packages/manager/src/features/CloudPulse/shared/CloudPulseDashboardFilterBuilder.tsx

```diff
@@ -120,17 +128,12 @@ export const CloudPulseDashboardFilterBuilder = React.memo(
     );
 
     const handleResourceChange = React.useCallback(
-      (resourceId: CloudPulseResources[], savePref: boolean = false) => {
-        emitFilterChangeByFilterKey(
-          RESOURCE_ID,
-          resourceId.map((resource) => resource.id),
-          savePref,
-          {
-            [RESOURCES]: resourceId.map((resource: { id: string }) =>
-              String(resource.id)
-            ),
-          }
-        );
+      (resources: CloudPulseResources[], savePref: boolean = false) => {
```

Comment:
Do we need to validate "resources" for null or undefined?
if (!Array.isArray(resources)) {
  throw new Error("Invalid input");
}